forge-game/src/main/java/forge/game/spellability/SpellAbility.java

```diff
@@ -937,36 +937,38 @@ public String getOriginalDescription() {
     public String getCostDescription() {
         if (payCosts == null || (this instanceof AbilitySub)) { // SubAbilities don't have Costs or Cost
             return "";
-        } else {
-            StringBuilder sb = new StringBuilder();
-            // descriptors
-            if (hasParam("PrecostDesc")) {
-                sb.append(getParam("PrecostDesc")).append(" ");
-            }
-            if (hasParam("CostDesc")) {
-                sb.append(getParam("CostDesc")).append(" ");
-            } else {
-                if (hasParam("AlternateCost")) {
-                    Cost alternateCost = new Cost(getParam("AlternateCost"), payCosts.isAbility());
-                    boolean altOnlyMana = alternateCost.isOnlyManaCost();
-                    if (payCosts.isOnlyManaCost() && !altOnlyMana) {
-                        sb.append("Pay ");
-                    }
-                    sb.append(payCosts.toString());
-                    sb.append(" or ").append(altOnlyMana ? alternateCost.toString() :
-                            StringUtils.uncapitalize(alternateCost.toString()));
-                    sb.append(isEquip() && !altOnlyMana ? "." : "");
-                } else {
-                    sb.append(payCosts.toString());
-                }
+        }
 
-                if (payCosts.isAbility() && !isEquip()) {
-                    sb.append(": ");
+        boolean equip = false;
+        StringBuilder sb = new StringBuilder();
+        // descriptors
+        if (hasParam("PrecostDesc")) {
+            equip = getParam("PrecostDesc").equals("Equip");
```

Comment:
`getParam("PrecostDesc").equals("Equip")` is too restrictive

from this part, it might be more?
```
abilityStr.append(" | PrecostDesc$ Equip");
if (k.length > 3 && !k[3].isEmpty()) {
    abilityStr.append(" ").append(vstr);
}
```

Reply:
good catch